lib/product.rb

```diff
@@ -0,0 +1,21 @@
+class Product
+  attr_reader :category, :name, :unit_price, :quantity
+  attr_accessor :hoard
```

Comment:
this should be part of the `attr_reader` as we are now allowing anything outside the product class to modify this hoard instance variable which we do not want